Lib/pathlib.py

```diff
@@ -583,6 +583,8 @@ def __len__(self):
             return len(self._parts)
 
     def __getitem__(self, idx):
+        if isinstance(idx, slice):
+            return [self[i] for i in range(*idx.indices(len(self)))]
```

Comment:
Since `PathLib.parents` returns an immutable sequence, slices should also be immutable, IMO, so this should return a tuple:

```suggestion
            return tuple(self[i] for i in range(*idx.indices(len(self))))
```